packages/devtools_app/test/flutter/timeline_screen_test.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/globals.dart';
+import 'package:devtools_app/src/service_manager.dart';
+import 'package:devtools_app/src/timeline/flutter/event_details.dart';
+import 'package:devtools_app/src/timeline/flutter/flutter_frames_chart.dart';
+import 'package:devtools_app/src/timeline/flutter/timeline_flame_chart.dart';
+import 'package:devtools_app/src/timeline/flutter/timeline_screen.dart';
+import 'package:devtools_app/src/timeline/timeline_controller.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../support/mocks.dart';
+import 'wrappers.dart';
+
+void main() {
+  TimelineScreen screen;
+  MockServiceManager mockServiceManager;
+
+  group('TimelineScreen', () {
+    setUp(() {
+      mockServiceManager = MockServiceManager();
+      setGlobal(ServiceConnectionManager, mockServiceManager);
+      when(serviceManager.connectedApp.isDartWebApp)
+          .thenAnswer((_) => Future.value(false));
+      screen = const TimelineScreen();
+    });
+
+    testWidgets('builds its tab', (WidgetTester tester) async {
+      await tester.pumpWidget(wrap(Builder(builder: screen.buildTab)));
+      expect(find.text('Timeline'), findsOneWidget);
+    });
+
+    testWidgets('builds proper content for timeline modes',
+        (WidgetTester tester) async {
+      // Set a wide enough screen width that we do not run into overflow.
+      await setWindowSize(const Size(1599.0, 1000.0));
+      var body = TimelineScreenBody();
+      await tester.pumpWidget(wrap(body));
+      expect(find.byType(TimelineScreenBody), findsOneWidget);
+      body = tester.widget(find.byType(TimelineScreenBody));
+
+      // Verify TimelineMode.frameBased content.
+      expect(body.controller.timelineMode, equals(TimelineMode.frameBased));
+      expect(find.text('Pause'), findsOneWidget);
+      expect(find.text('Resume'), findsOneWidget);
+      expect(find.text('Record'), findsNothing);
+      expect(find.text('Stop'), findsNothing);
+      expect(find.byType(FlutterFramesChart), findsOneWidget);
+      expect(find.byType(TimelineFlameChart), findsOneWidget);
+      expect(find.byType(EventDetails), findsOneWidget);
+
+      // Switch timeline mode and re-pump widget.
+      body = TimelineScreenBody()..controller.timelineMode = TimelineMode.full;
```

Comment:
If you want to just work directly with screen instead of changing the controller, you can also use `tester.tap` on the dropdown to select what you want.